ringbuffer.py

```diff
@@ -185,23 +189,29 @@ def try_write(self, data):
             self.array[position.index] = data
             self.writer.increment()
 
-    def _has_read_conflict(self, reader_position):
+    def _try_read_no_lock(self, reader, length: int = 1):
+        position = reader.get()
         writer_position = self.writer.get()
-        return writer_position.counter <= reader_position.counter
 
-    def _try_read_no_lock(self, reader):
-        position = reader.get()
-        if self._has_read_conflict(position):
+        if writer_position.counter <= position.counter:
             if not self.active.value:
                 raise WriterFinishedError
             else:
                 raise WaitingForWriterError
 
-        data = self.array[position.index]
-        reader.increment()
-        return data
+        max_length = writer_position.counter - position.counter
+
+        if max_length < length:
+            raise NotEnoughItemsError
+
+        new_array = (self.c_type * length)()
+        ctypes.memmove(new_array,
+                       ctypes.addressof(self.array[position.index]),
+                       ctypes.sizeof(new_array))
+        reader.increment_by(length)
```

Comment:
Nice!

Should we document somewhere that the `ctypes` struct shouldn't contain internal pointers that might be invalidated here though (or really pointers in general, because I think the semantics get wonky there)? I can't really imagine that we'd every use one of those, but it'd be good to note it just in case.

Reply:
Yes, I'm going to add docs about that.